### PR TITLE
Add OCaml example for brainfuck

### PR DESCRIPTION
**Pull request by melwyn95**

### CODE REVIEW

**Comment by eatonphil** [comment]
Hey! This is a good first start! But the rules say not to switch algorithm or paradigm even to be more idiomatic.

Since ocaml supports mutable struct and values, I'd like the example to use that too.

Feel free to make that change if you'd like or I can make it later in.

Thanks and good start!

File: .github/workflows/pull_requests.yml [hunk 1]
```diff
@@ -17,4 +17,10 @@ jobs:
       with:
         version: master
 
+    # Set up OCaml
+    - name: Use OCaml 4.14.x
+    uses: ocaml/setup-ocaml@v2
```
Comment (eatonphil) [inline .github/workflows/pull_requests.yml]: The indentation here is wrong

Comment (melwyn95) [reply .github/workflows/pull_requests.yml]: Nice catch, will fix this in the next commit

**Comment by melwyn95** [comment]
> Since ocaml supports mutable struct and values, I'd like the example to use that too.

Thanks for having a look, I'll make this change :)

File: .github/workflows/pull_requests.yml (hunk 1 above)
Comment (melwyn95) [reply .github/workflows/pull_requests.yml]: Fixed

**Comment by melwyn95** [comment]
I kept the functional example from the last commit & added a new example using imperative features of OCaml

Can we keep both examples? if no then let me know I'll remove the example using functional paradigm.

Thanks :)

**Comment by eatonphil** [comment]
Ok this looks pretty good! Yes we can keep both.

File: Brainfuck/ocaml/program.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+prepare:
+  - ocamlc bf.functional.ml -o bf.functional
+  - ocamlc bf.imperative.ml -o bf.imperative
+run:
+  - ./bf.functional {PROGRAM}
+  - ./bf.imperative {PROGRAM}
```
Comment (eatonphil) [inline Brainfuck/ocaml/program.yaml:6]: I think the tests right now depend on there being a single `run`.

Comment (melwyn95) [reply Brainfuck/ocaml/program.yaml:6]: Ahh! I'll handle that in the test script

Comment (melwyn95) [reply Brainfuck/ocaml/program.yaml:6]: Done, can you have a look?

Comment (eatonphil) [reply Brainfuck/ocaml/program.yaml:6]: Updated it slightly after a merge conflict but yeah looks ok now

**Comment by eatonphil** [comment]
This looks like it's failing because you aren't installing/setting up ocaml correctly: `/bin/sh: 1: ocamlc: not found`

File: scripts/test.py [hunk 3]
```diff
@@ -23,16 +23,20 @@ def subdirs(path):
...
-        print("[{}, {}]: Expected '{}', got '{}'.".format(project.split('/')[1], language.split('/')[-1], expected, out))
-        global failed
-        failed = True
+        
+    run_steps = program['run']
+    run_steps = run_steps if isinstance(run_steps, list) else [run_steps]
+
+    for run_step in run_steps:
+        try:
+            out = subprocess.check_output(program['run'].format(PROGRAM="../_tests/" + test), shell=True, cwd=language)
```
Comment (melwyn95) [inline scripts/test.py]: I think this should be
```suggestion
            out = subprocess.check_output(run_step .format(PROGRAM="../_tests/" + test), shell=True, cwd=language)
```

Comment (eatonphil) [reply scripts/test.py]: Fixed. Not taking your change since you put a whitespace in there between run_step and .format 😂

Comment (melwyn95) [reply scripts/test.py]: lol, sorry

**Comment by melwyn95** [comment]
> This looks like it's failing because you aren't installing/setting up ocaml correctly: /bin/sh: 1: ocamlc: not found

I'll fix this tomorrow

**Comment by eatonphil** [comment]
Nice! Looks good!

**Pull request merged**